internal/providers/msentraid/himmelblau.go

```diff
@@ -0,0 +1,372 @@
+		attrs,
+		tpm,
+		machineKey,
+		&cTransportKey,
+		&cCertKey,
+		&cDeviceID,
+	)
+	if msalErr != nil {
+		return nil, fmt.Errorf("failed to enroll device: %v", C.GoString(msalErr.msg))
+	}
+
+	deviceID := C.GoString(cDeviceID)
+	log.Infof(ctx, "Enrolled device with ID: %v", deviceID)
+
+	var certKey []byte
+	var cSerializedCertKey *C.char
+	var cSerializedCertKeyLen C.size_t
+	defer C.free(unsafe.Pointer(cSerializedCertKey))
+	msalErr = C.serialize_loadable_ms_device_enrolment_key(cCertKey, &cSerializedCertKey, &cSerializedCertKeyLen)
+	if msalErr != nil {
+		return nil, fmt.Errorf("failed to serialize device enrollment key: %v", C.GoString(msalErr.msg))
+	}
+	if cSerializedCertKeyLen > 0 {
+		certKey = C.GoBytes(unsafe.Pointer(cSerializedCertKey), C.int(cSerializedCertKeyLen))
+	}
+
+	var transportKey []byte
+	var cSerializedTransportKey *C.char
+	var cSerializedTransportKeyLen C.size_t
+	defer C.free(unsafe.Pointer(cSerializedTransportKey))
+	msalErr = C.serialize_loadable_ms_oapxbc_rsa_key(cTransportKey, &cSerializedTransportKey, &cSerializedTransportKeyLen)
+	if msalErr != nil {
+		return nil, fmt.Errorf("failed to serialize transport key: %v", C.GoString(msalErr.msg))
+	}
+	if cSerializedTransportKeyLen > 0 {
+		transportKey = C.GoBytes(unsafe.Pointer(cSerializedTransportKey), C.int(cSerializedTransportKeyLen))
+	}
+
+	var tpmMachineKey []byte
+	var cSerializedTpmMachineKey *C.char
+	var cSerializedTpmMachineKeyLen C.size_t
+	defer C.free(unsafe.Pointer(cSerializedTpmMachineKey))
+	msalErr = C.serialize_loadable_machine_key(loadableMachineKey, &cSerializedTpmMachineKey, &cSerializedTpmMachineKeyLen)
+	if msalErr != nil {
+		return nil, fmt.Errorf("failed to serialize TPM machine key: %v", C.GoString(msalErr.msg))
+	}
+	if cSerializedTpmMachineKeyLen > 0 {
+		tpmMachineKey = C.GoBytes(unsafe.Pointer(cSerializedTpmMachineKey), C.int(cSerializedTpmMachineKeyLen))
+	}
+
+	jsonData, err := json.Marshal(deviceRegistrationData{
+		DeviceID:      deviceID,
+		CertKey:       certKey,
+		TransportKey:  transportKey,
+		AuthValue:     C.GoString(cAuthValue),
+		TPMMachineKey: tpmMachineKey,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal device registration data: %v", err)
+	}
+
+	return jsonData, nil
+}
+
+func hostname() string {
+	name, err := os.Hostname()
+	if err != nil {
+		return "unknown"
+	}
+	return name
+}
+
+// OSVersion gets the pretty name of the OS release from the system.
+// Since we're running in a snap, this returns the version of the core base snap
+// (which is not that helpful when it's shown as the device's OS in Entra, so
+// might want to change this in the future, to somehow get the host's OS version).
+func OSVersion() string {
+	data, err := os.ReadFile("/etc/os-release")
+	if err != nil {
+		return "unknown"
+	}
+
+	for _, line := range strings.Split(string(data), "\n") {
+		if strings.HasPrefix(line, "PRETTY_NAME=") {
+			return strings.Trim(strings.TrimPrefix(line, "PRETTY_NAME="), "\"")
+		}
```

Comment:
```suggestion
		if name, found := strings.CutPrefix(line, "PRETTY_NAME=")); found {
			return name
		}
```

Then if you want to use the happy path, would be in an extra line.